### PR TITLE
Update EditorConfig and Git files

### PR DESCRIPTION
**Pull request by Nirmal4G**
## Info

Extracted from CommunityToolkit/WindowsCommunityToolkit#4234 as the Common Code projects were separated in this new repository.

## Changes

- Remove BOM from all files.

- Add default Git attributes.
- Update Git ignores to latest.
- Merge duplicate EditorConfig rules.
- Format projects to EditorConfig rules.
- Fix formatting errors in 'Guard.md' file.

## PR Checklist

- [x] Created a `feature` branch in your fork
- [x] Based off latest main branch of toolkit
- [x] PR doesn't include merge commits (**always rebase on top of our main**)
- [x] Header has been added to all new source files (ran ***build/Update-Headers.ps1***)
- [x] Contains **NO** breaking changes
- [x] Code follows all style conventions

## Notes

- **Rebase** merge please.
- When merging, please update the commit title to PR title instead of the default `Merge pull request #xxxx from repo/branch`, and commit message to either PR message or messages of individual commits. The `auto-merge` option does this by default.

### CODE REVIEW

**Comment by Nirmal4G** [comment]
@Sergio0694 I have extracted non-code changes here. This should be merged first.

File: CommunityToolkit.Diagnostics/Generated/Guard.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
- - In some cases, using generic methods just isn't possible. For instance, types like `Span<T>` and `ReadOnlySpan<T>` can't be used as generic type parameters, and even if that had been possible, they don't implement an interface we could use in the generic type constraint. Using T4 templates solves this issue, as we can just have specialized method for each supported type or collection type.
+- Especially when many overloads are available for an API, using templates makes it much easier to maintain code and spot mistakes, as the actual number of lines of code to review is much smaller: it's just the same code from each template!
+- Using type-specific overloads instead of generic methods can result in faster code. For instance, T4 templates are used to generate overloads for comparison APIs (e.g. `Guard.IsGreaterThan(int, int, string)`). This results in more compact and optimized code as opposed to a generic method using `where T : IComparable<T>` as type constraint.
```
Comment (Sergio0694) [inline CommunityToolkit.Diagnostics/Generated/Guard.md:10]: Please leave "eg." spelled that way, it's the same we're using everywhere else as well.
(squash rebase when you fix this, don't add a new commit just for that).

**Review by Sergio0694 — APPROVED** [review]
This was much easier to review, thank you!
Question: why do you want this to be a rebase merge and not a normal merge like other PRs?

File: CommunityToolkit.Diagnostics/Generated/Guard.md (hunk 1 above)
Comment (Nirmal4G) [reply CommunityToolkit.Diagnostics/Generated/Guard.md:10]: The spell checker complains about it often. Are you sure?

Comment (Sergio0694) [reply CommunityToolkit.Diagnostics/Generated/Guard.md:10]: Yeah, we have "eg." in dozens of places all over the repo. I personally think it's fine, but even if not, then we should just update all of those occurrences together separately, not here. But really I don't think it's an issue at all to be honest.

**Comment by Nirmal4G** [comment]
> Why do you want this to be a rebase merge and not a normal merge like other PRs?

It contains two different changes, that's why. Same for my other PRs. Although they depend on some changes in the previous PR/commit, overall, the changes in every commit are standalone.

File: CommunityToolkit.Diagnostics/Generated/Guard.md (hunk 1 above)
Comment (Nirmal4G) [reply CommunityToolkit.Diagnostics/Generated/Guard.md]: Done. I'll open a separate PR for spelling fixes.

**Comment by Nirmal4G** [comment]
@Sergio0694 All done

**Comment by Sergio0694** [comment]
Thanks! 🙂

**Pull request merged**